Notebooks/verify_truth.ipynb

```diff
@@ -0,0 +1,724 @@
+   "metadata": {},
+   "source": [
+    "Convenience utilities"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_pg_columns(catalog, select_list, alias_list=None, other=None):\n",
+    "    '''\n",
+    "    Returns data frame version of db select output.  \n",
+    "    Works for PostgreSQL.  It seems sqlite cursors don't support \"with\"\n",
+    "    '''\n",
+    "    cols = ','.join(select_list)\n",
+    "    q = f\"\"\"SELECT {cols} FROM {catalog.pg_table}\"\"\"\n",
+    "    if other:\n",
+    "        q = ''.join([q,' ',other])\n",
+    "        \n",
+    "    #print('Postgres query: ', q)\n",
+    "    conn = Catalog.pg_connection()\n",
+    "\n",
+    "    with conn.cursor() as cursor:\n",
+    "        cursor.execute(q)\n",
+    "        records = cursor.fetchall()\n",
+    "\n",
+    "    if alias_list:\n",
+    "        return pd.DataFrame(records, columns=alias_list)\n",
+    "    else:\n",
+    "        return pd.DataFrame(records, columns=select_list)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_sqlite_columns(catalog, select_list, alias_list=None, other=None):\n",
+    "    '''\n",
+    "    Returns data frame version of db select output.  \n",
+    "    '''\n",
+    "    cols = ','.join(select_list)\n",
+    "    q = f\"\"\"SELECT {cols} FROM {catalog.sqlite_table}\"\"\"\n",
+    "    if other:\n",
+    "        q = ''.join([q,' ',other])\n",
+    "    conn = sqlite3.connect(catalog.sqlite_abspath)\n",
+    "    cursor = conn.cursor()\n",
+    "  \n",
+    "    cursor.execute(q)\n",
+    "    records = cursor.fetchall()\n",
+    " \n",
+    "    if alias_list:\n",
+    "        return pd.DataFrame(records, columns=alias_list)\n",
+    "    else:\n",
+    "        return pd.DataFrame(records, columns=select_list)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_parquet_columns(catalog, column_list):\n",
+    "    return(pq.read_table(catalog.parquet_abspath, column_list)).to_pandas()\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Standard queries, to be applied to each representation should include\n",
+    "* getting length of id column (all tables have an id column)\n",
+    "* comparing values for a couple rows.  For sqlite and parquet, expect rows will be returned in the same order.  For PostgreSQL might have to match ids (for variability tables would have to match (id, obshistid) for uniqueness)\n",
+    "* plotting ra, dec (summary tables)\n",
+    "* for variability tables, plot a light curve\n",
+    "* histogramming a column or two"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Compare counts\n",
```

Comment:
This cell takes quite a while to run. Add a warning for the user?

Reply:
Good idea.